crypto/src/main/scala/coop/rchain/crypto/util/CertificateHelper.scala

```diff
@@ -110,14 +120,65 @@ object CertificateHelper {
     val cert = new X509CertImpl(info)
     cert.sign(privateKey, algorythm)
 
-    // Update the algorith, and resign.
+    // Update the algorithm, and resign.
     val algorithmId2 = cert.get(X509CertImpl.SIG_ALG).asInstanceOf[AlgorithmId]
     info.set(CertificateAlgorithmId.NAME + "." + CertificateAlgorithmId.ALGORITHM, algorithmId2)
     val cert2 = new X509CertImpl(info)
     cert2.sign(privateKey, algorythm)
     cert2
   }
 
+  def encodeSignatureRStoDER(signatureRS: Array[Byte]): Either[Throwable, Array[Byte]] = {
+    def toASN1Int(bytes: Array[Byte]) = new ASN1Integer(BigIntegers.fromUnsignedByteArray(bytes))
+
+    def convert = {
+      val (r, s) = signatureRS.take(64).splitAt(32)
+      val bos    = new ByteArrayOutputStream(72)
+      val seq    = new DERSequenceGenerator(bos)
+      try {
+        seq.addObject(toASN1Int(r))
+        seq.addObject(toASN1Int(s))
+        seq.close
+        bos.toByteArray
+      } finally {
+        if (seq != null) seq.close()
+        // > Closing a ByteArrayOutputStream has no effect.
+        // https://docs.oracle.com/javase/10/docs/api/java/io/ByteArrayOutputStream.html
+      }
+    }
+
+    if (signatureRS.isEmpty)
+      new Exception("Input array must not be empty.").asLeft
+    else
+      Try(convert).toEither
+  }
+
+  def decodeSignatureDERtoRS(signatureDER: Array[Byte]): Either[Throwable, Array[Byte]] = {
+    def toBytes(x: ASN1Encodable) = {
+      val asn1 = x.toASN1Primitive.asInstanceOf[ASN1Integer]
+      BigIntegers.asUnsignedByteArray(asn1.getValue)
+    }
+
+    def convert: Array[Byte] = {
+      val bis = new ByteArrayInputStream(signatureDER)
+      val asn = new ASN1InputStream(bis)
+      try {
+        val asnSeq          = asn.readObject.asInstanceOf[ASN1Sequence]
+        val Array(r, s, _*) = asnSeq.toArray
+        toBytes(r) ++ toBytes(s)
+      } finally {
+        if (asn != null) asn.close()
+        // > Closing a ByteArrayInputStream has no effect.
+        // https://docs.oracle.com/javase/10/docs/api/java/io/ByteArrayInputStream.html
```

Comment:
This is true, but, being paranoid, what if someone changes the type of `InputStream` to something that needs to be closed?

Reply:
@marcin-rzeznicki maybe also to wrap `Try(asn.close)`?